static/js/containers/DashboardPage.js

```diff
@@ -294,6 +297,17 @@ class DashboardPage extends React.Component {
     this.fetchCoupons()
     this.handleOrderStatus()
     this.fetchDiscussionsFrontpage()
+    this.fetchUserProfile()
+  }
+
+  fetchUserProfile = () => {
+    const { financialAid: { fetchError }, dispatch } = this.props
+    if (fetchError && fetchError.message === "Profile is not complete") {
+      dispatch(clearProfile(SETTINGS.user.username))
```

Comment:
what's the reason for clearing the profile here?

Reply:
because if i dont then it will load missing data from cache.
This is for my use case when i delete users profile data from console and do not refresh dashboard and just submit income

Reply:
Ok I see, how likely do you think it is that a user will end up in the same position?

Reply:
i think that is the case, it is a bug in system which leads use to submit income with completing profile. This check is important

Reply:
Ok, I tested it out and it seems to work correctly. FYI the most likely thing that allowed a user to end up in this kind of state (a bug in our profile redirect behavior on the frontend) was fixed in #3900, so hopefully we won't run into it again. But I think what you said makes sense, and we should keep the `clearProfile` call here